torch/_inductor/scheduler.py

```diff
@@ -41,6 +52,27 @@
 fusion_log = torch._logging.getArtifactLogger(__name__, "fusion")
 
 
+class WhyNoFuse:
```

Comment:
Any reason why this shouldn't just be a dataclass? With slots=True?

Reply:
that was my initial attempt :) but I realized that `slots=True` is only supported in Python >= 3.10

Reply:
Hmm, so apparently you can still use a dataclass, but define slots yourself as long as you don't use default args in them. But yeah maybe just add a comment about why so we can fix it when we drop support for 3.9